src/main/java/edu/cornell/kfs/module/purap/rest/resource/CuEinvoiceApiResource.java

```diff
@@ -0,0 +1,248 @@
+            safelyAddProperty(vendorProperties, CUPurapConstants.Einvoice.ADDRESS_LINE2, vendorAddress.getVendorLine2Address());
+            safelyAddProperty(vendorProperties, CUPurapConstants.Einvoice.CITY, vendorAddress.getVendorCityName());
+            safelyAddProperty(vendorProperties, CUPurapConstants.Einvoice.STATE, vendorAddress.getVendorStateCode());
+            safelyAddProperty(vendorProperties, CUPurapConstants.Einvoice.ZIPCODE, vendorAddress.getVendorZipCode());
+            safelyAddProperty(vendorProperties, CUPurapConstants.Einvoice.COUNTRY, vendorAddress.getVendorCountryCode());
+            safelyAddProperty(vendorProperties, CUPurapConstants.Einvoice.ADDRESS_TYPE_CODE, vendorAddress.getVendorAddressTypeCode());
+        }
+    }
+
+    private VendorAddress getVendorAddress(List<VendorAddress> vendorAddresses) {
+        for (VendorAddress address : vendorAddresses) {
+            if (address.getVendorAddressType().getVendorAddressTypeCode().equalsIgnoreCase(VendorConstants.AddressTypes.REMIT)) {
+                return address;
+            }
+        }
+        return vendorAddresses.get(0);
+    }
+
+    private void addPoItemsToProperties(Properties poProperties, List poItems) {
+        List<Properties> poLines = new ArrayList<>();
+        for (Object obj : poItems) {
+            if (obj instanceof PurApItem) {
+                PurApItem purApItem = (PurApItem)obj;
+                Properties lineProps = new Properties();
+                safelyAddProperty(lineProps, CUPurapConstants.AMOUNT.toLowerCase(), purApItem.getTotalAmount().toString());
+                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.QUANTITY, purApItem.getItemQuantity().toString());
+                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.UNIT_OF_MEASURE, purApItem.getItemUnitOfMeasureCode());
+                safelyAddProperty(lineProps, CUPurapConstants.Einvoice.DESCRIPTION, purApItem.getItemDescription());
+                lineProps.put(CUPurapConstants.Einvoice.UNIT_PRICE, purApItem.getItemUnitPrice());
+                lineProps.put(CUPurapConstants.Einvoice.LINE_NUMBER, purApItem.getItemLineNumber());
+                lineProps.put(CUPurapConstants.Einvoice.PART_NUMBER, purApItem.getItemAuxiliaryPartIdentifier());
+                poLines.add(lineProps);
+            }
+        }
+        poProperties.put(CUPurapConstants.Einvoice.ITEMS, poLines.toArray());
+    }
+
+    private String getSimpleJsonObject(String key, String value) {
+        Properties obj = new Properties();
+        obj.put(key, value);
+        return gson.toJson(obj);
+    }
+
+    private LookupDao getLookupDao() {
+        if (ObjectUtils.isNull(lookupDao)) {
+            lookupDao = SpringContext.getBean(LookupDao.class);
+        }
+        return lookupDao;
+    }
+
+    private CuEinvoiceDao getCuEinvoiceDao() {
+        if (ObjectUtils.isNull(null)) {
```

Comment:
This will always return true, did you want cuEinvoiceDao instead of null here

Reply:
Thanks for catching this bug